streams/src/test/java/org/apache/kafka/streams/tests/StreamsUpgradeTest.java

```diff
@@ -66,4 +98,150 @@ public void run() {
             }
         });
     }
+
+    private static class FutureKafkaClientSupplier extends DefaultKafkaClientSupplier {
```

Comment:
Are these going to be thrown away code when we bump the version again in the future releases?

Reply:
I think we should keep. Even if we bump to version 4, we should test is we can upgrade to version 5.